packages/mui-utils/src/useUndoAndRedo.ts

```diff
@@ -0,0 +1,55 @@
+import isEqual from 'lodash/isEqual';
+import { useState } from 'react';
+
+const useUndoAndRedo = <T>(initialValue?: T) => {
+  const [values, setValues] = useState<T[]>(initialValue !== undefined ? [initialValue] : []);
+  const [index, setIndex] = useState(initialValue !== undefined ? 0 : -1);
```

Comment:
What do you think about having a single state? It seems that both properties are intrinsically linked, e.g. how it's done in https://github.com/react-simple-code-editor/react-simple-code-editor/blob/4065fa6da6b7ca5eefc660b151fe5ab8b1a740cf/src/index.tsx#L499